### PR TITLE
Hotfix for sync error due to missing concepts

### PR DESCRIPTION
**Pull request by koushik-ms**
Hotfix for #1323 
This PR creates placeholder documentation for the missing concepts causing the sync-failure.

### CODE REVIEW

**Review by coriolinus — COMMENTED** [review]
As a matter of policy, we're trying to keep TODO items out of the main branch. That implies one of a few different futures:

- If you have the time and inclination to write a substantial amount of documentation, fleshing out all the TODOs here would be great.
- Otherwise, we can close this PR and open a new one getting rid of the [recently added concepts](https://github.com/exercism/rust/pull/1322/files#diff-587cb980af76fdc7e52369fd0b9d926dff266976b6f8ac631e358fecc49ff8cfR1524-R1562).
- Otherwise, we can close this PR and revert #1322 entirely.

Obviously the first option is strongly preferable from my perspective, but that's only viable if you're willing to tackle the task. Let me know which of these seems preferable to you.

**Comment by koushik-ms** [comment]
I attempted to do this for loops concept. Since this is the first time I do this, I would appreciate some feedback. If this is in the right direction, I would like to continue with the rest else I am happy to do a rollback.

File: concepts/loops/about.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+The above code-snippet makes an infinite loop. To exit a loop use the break keyword:
+
+```rust
+fn main() {
+    let mut count = 0;
+    loop {
+        println!("again!");
+        count += 1;
+        if count > 10 { break ; }
```
Comment (coriolinus) [inline concepts/loops/about.md:23]:
```suggestion
        if count > 10 { break; }
```

File: concepts/loops/about.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    }
+}
+```
+
+## Loops using `for` and `while`
+
+Apart from the `loop` keyword, you can also make loops with `for` and `while`.
+
+A `for` loop is used to iterate over the elements of a collection.
```
Comment (coriolinus) [inline concepts/loops/about.md:32]:
```suggestion
A `for` loop is used to iterate over the elements of a collection or iterator.
```

Comment (coriolinus) [reply concepts/loops/about.md:32]: Might be worth adding a sentence explaining that ranges are iterators as well, so you can do things like `for i in 0..10 { ... }`.

File: concepts/loops/introduction.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Introduction
+
+Loops enable repetition of blocks of code. There are three types of loops in Rust: the `for` loop that enables looping over elements of a collection, the `while` loop that repeates a block of code as long as a condition is satisfied and the `loop` loop that indefinitely repeats code until a `break` is invoked explicitly.
```
Comment (coriolinus) [inline concepts/loops/introduction.md:3]: Same here: `for` loops fundamentally act on anything implementing `IntoIterator`. Collections have that property, but so do iterators and certain other special-case types.

**Review by coriolinus — COMMENTED** [review]
Sorry for the delay in review. This is great work, and I appreciate you contributing it! If you're willing to do the rest of the topics in the same style, that would be extremely helpful.

File: concepts/loops/about.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+fn main() {
+    let a = [10, 20, 30, 40, 50];
+
+    for x in a {
+        println!("the value is: {}", x);
+    }
+}
+```
+
+Ranges are itearators too, which makes itearating through them with the `for` loop very convenient:
```
Comment (coriolinus) [inline concepts/loops/about.md:44]:
```suggestion
Ranges are iterators too, which makes itearating through them with the `for` loop very convenient:
```

typo

**Comment by coriolinus** [comment]
Loops stuff looks great!

File: concepts/loops/about.md (hunk 4 above)
Comment (koushik-ms) [reply concepts/loops/about.md]: itear-apart-iterators!

File: concepts/hashmap/introduction.md [hunk 5]
```diff
@@ -1,3 +1,3 @@
 # Introduction
 
-TODO
+Hashmap is a collection used to store key-value pairs. It allows arbitrary and user-defined types to be used as keys to index into it (technically any type that implement `Eq` and `Hash` traits can be used as keys). Hashmap have fast insertion and retrieval functions and are very useful in e.g., making frequency counts, storing relationships etc.,. Hashmap also features the powerful `entry` API that makes it ergonomic to use.
```
Comment (coriolinus) [inline concepts/hashmap/introduction.md:3]:
```suggestion
Hashmap is a collection used to store key-value pairs. It allows arbitrary types to be used as keys and values. Hashmap have fast insertion and retrieval functions and are very useful in e.g., making frequency counts, storing relationships etc.,. Hashmap also features the powerful `entry` API that makes it ergonomic to use.
```

- arbitrary is a superset of user-defined
- arbitrary keys _and values_
- `Eq` and `Hash` bounds should be mentioned in `about`, not `introduction`

File: concepts/hashmap/about.md [hunk 6]
```diff
@@ -1,3 +1,61 @@
 # Hashmap
 
-TODO
+Hashmap is a container that can be used to store "key-value pairs". This container is also known by the name associative array or Hashtable in other programming langauges. In short, there is a value stored in the hashmap and it is mapped to a key. The user can retrieve the stored value by passing in the key or store a value mapped to a specific key.
+
+Hashmap is a generic collection (like most of the collections in the standard library) and thus supports a broad variety of types as keys (including user-defined structs and tuples).
+
+# Creating a Hashmap
```
Comment (coriolinus) [inline concepts/hashmap/about.md:7]:
```suggestion
## Creating a Hashmap
```

Sub-headers should be level 2, not level 1.

**Review by coriolinus — COMMENTED** [review]
Hashmap concept documentation looking great so far! A few notes:

- the `Eq` and `Hash` bounds for keys should be explained in `about.md`, probably in a `## Trait Bounds` section 
- The same section should mention that there is no trait bound for values
- There should be a `## Efficiency` section in `about.md` which explains that hashmaps have amortized `O(1)` for insert, check existence, retrieve, update, and delete.

**Comment by coriolinus** [comment]
This PR has stalled. No shame in that; it was very large! To spur further progress, I've broken out the work into several follow-on PRs:

- #1345 removes the new concepts which were causing the sync errors
- #1346 has the hashmap concept portion of this PR
- #1347 has the loops concept portion of this PR
- #1348 has the match-basics portion of this PR
- #1349 has the mutability portion of this PR
- #1350 has the string portion of this PR
- #1351 has the vec-macro portion of this PR
- #1352 has the vec-stack portion of this PR

With any luck, it'll be easier to make progress on those individual PRs instead of this huge one.

**Pull request closed without merging**